### ISSUE TITLE
fmt should be able to flush stdin before printing the prompt and Scanln

### ISSUE THREAD

**Issue by akevinbailey** [golang/go#13210]
Go needs a way to flush the stdin before prompt and Scanln.

See <a href="https://github.com/odeke-em/drive/issues/157">https://github.com/odeke-em/drive/issues/157</a> for psoible fix for Linux.

**Comment by adg** [comment]
This isn't a `fmt` issue. It's a feature request for a function that should probably be in the `x/sys` repo.

This is how @minux implemented it, by calling out to a C library: https://github.com/minux/drive/blob/2b076a3dc02b5fee728e9eeee8c089a414de5fbc/src/tty.go

**Comment by akevinbailey** [comment]
I disagree.  Flushing stdio should be a basic function of GO without having to call an external C library.  Java and Python have it built in.

**Comment by ianlancetaylor** [comment]
How do you flush standard input in Java and Python?  Can you point to the docs?

Note that this issue is not about flushing standard output.  Go can already do that perfectly well.

**Comment by akevinbailey** [comment]
Sorry, I meant stdin

Java:
    System.in.read(new byte[System.in.available()]);  // Flushes stdin

Python:
      from termios import tcflush, TCIOFLUSH
      tcflush(sys.stdin, TCIOFLUSH)

**Comment by adg** [comment]
On 12 November 2015 at 12:18, A. Kevin Bailey <notifications@github.com>
wrote:

> from termios import tcflush, TCIOFLUSH
> tcflush(sys.stdin, TCIOFLUSH)
>

This is the kind of thing I was suggesting in renaming this issue.

**Comment by tklauser** [comment]
Some of the `termios.h` function require more functionality than just mere syscall wrappers (which is the scope of the `x/sys` package) or they don't use any syscalls at all (like the `cf*` functions). Thus, it might be more appropriate to implement these functions in the (currently empty) `x/term` package? We may then make use of `x/sys` for certain parts of the implementation.

Also, some parts like `cfmakeraw` are already available in `x/crypto/ssh/terminal` and I guess we should move/copy them to `x/term` as well.
